### ISSUE TITLE
SQL Syntax error submitting a form

### ISSUE THREAD

**Issue by DieterVanhove** [verbb/formie#2138]
### Describe the bug

I recently upgraded my project from craft cms 4 to 5.
I'm encountering a SQL syntax error when trying to submit a form. This error occurs during the execution of an ElementQuery that appears to be joining multiple tables (elements, elements_sites, structureelements, entries). The error seems to be related to an incomplete or malformed SQL ORDER BY clause within a subquery.

I traced error back to the `_setTitle` function inside the `SubmissionsController` on line
`$submission->title = Variables::getParsedValue($form->settings->submissionTitleFormat, $submission, $form);`

Please let me know if you need more information.

### Steps to reproduce

1. submit existing form
2. exception is thrown explained above

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: Yes

### Craft CMS version

5.4.10.1

### Plugin version

dev-craft-5 as 3.0.9

### Multi-site?

yes

### Additional context

Stack trace:
```
2024-11-12 15:52:04 [web.ERROR] [yii\db\Exception] PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`' at line 10 in /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php:1320
Stack trace:
#0 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php(1320): PDOStatement->execute()
#1 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php(1186): yii\db\Command->internalExecute('SELECT `element...')
#2 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php(417): yii\db\Command->queryInternal('fetchAll', NULL)
#3 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Query.php(249): yii\db\Command->queryAll()
#4 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/db/Query.php(292): yii\db\Query->all(NULL)
#5 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1853): craft\db\Query->all(NULL)
#6 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Extension/CoreExtension.php(1802): craft\elements\db\ElementQuery->all()
#7 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/helpers/Template.php(148): Twig\Extension\CoreExtension::getAttribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\elements\db\EntryQuery), 'all', Array, 'method', false, false, false, 1)
#8 /Users/dietervanhove/Sites/jims/storage/runtime/compiled_templates/f7/f7b3a30eff1a740b17a202a0c7789211.php(43): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\elements\db\EntryQuery), 'all', Array, 'method', false, false, false, 1)
#9 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Template.php(393): __TwigTemplate_8077349800ff7de111a57c341166c022->doDisplay(Array, Array)
#10 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Template.php(349): Twig\Template->yield(Array, Array)
#11 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Template.php(364): Twig\Template->display(Array)
#12 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/TemplateWrapper.php(35): Twig\Template->render(Array)
#13 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Environment.php(306): Twig\TemplateWrapper->render(Array)
#14 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/View.php(500): Twig\Environment->render('formie/_special...', Array)
#15 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/models/Notification.php(166): craft\web\View->renderTemplate('formie/_special...', Array, 'cp')
#16 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/base/Field.php(1339): verbb\formie\models\Notification->renderTemplate(Array, Array)
#17 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/base/Field.php(578): verbb\formie\base\Field->getEmailHtml(Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification), Object(craft\elements\db\EntryQuery))
#18 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/helpers/Variables.php(357): verbb\formie\base\Field->getValueForVariable(Object(craft\elements\db\EntryQuery), Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification))
#19 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/helpers/Variables.php(343): verbb\formie\helpers\Variables::getParsedFieldValue(Object(verbb\formie\fields\Entries), Object(craft\elements\db\EntryQuery), Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification))
#20 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/helpers/Variables.php(241): verbb\formie\helpers\Variables::getParsedFieldValues(Object(verbb\formie\elements\Form), Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification))
#21 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/controllers/SubmissionsController.php(1168): verbb\formie\helpers\Variables::getParsedValue('{timestamp}', Object(verbb\formie\elements\Submission), Object(verbb\formie\elements\Form))
#22 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/controllers/SubmissionsController.php(1136): verbb\formie\controllers\SubmissionsController->_setTitle(Object(verbb\formie\elements\Submission), Object(verbb\formie\elements\Form))
#23 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/controllers/SubmissionsController.php(438): verbb\formie\controllers\SubmissionsController->_populateSubmission(Object(verbb\formie\elements\Form))
#24 [internal function]: verbb\formie\controllers\SubmissionsController->actionSubmit()
#25 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#26 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#27 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('submit', Array)
#28 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/Application.php(350): yii\base\Module->runAction('formie/submissi...', Array)
#29 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/Application.php(649): craft\web\Application->runAction('formie/submissi...', Array)
#30 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/Application.php(312): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#31 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#32 /Users/dietervanhove/Sites/jims/web/index.php(17): yii\base\Application->run()
#33 /Users/dietervanhove/.composer/vendor/laravel/valet/server.php(110): require('/Users/dieterva...')
#34 {main}

Next yii\db\Exception: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`' at line 10
The SQL being executed was: SELECT `elements`.`id`, `elements`.`canonicalId`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateLastMerged`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`id` AS `siteSettingsId`, `elements_sites`.`siteId`, `elements_sites`.`title`, `elements_sites`.`slug`, `elements_sites`.`uri`, `elements_sites`.`content`, `elements_sites`.`enabled` AS `enabledForSite`, `entries`.`sectionId`, `entries`.`fieldId`, `entries`.`primaryOwnerId`, `entries`.`typeId`, `entries`.`postDate`, `entries`.`expiryDate`, `structureelements`.`root`, `structureelements`.`lft`, `structureelements`.`rgt`, `structureelements`.`level`, `structureelements`.`structureId`
FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `siteSettingsId`, `structureelements`.`structureId`
FROM `elements` `elements`
INNER JOIN `entries` `entries` ON `entries`.`id` = `elements`.`id`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `elements`.`id`) AND (EXISTS (SELECT *
FROM `structures` use index(primary)
WHERE (`id` = `structureelements`.`structureId`) AND (`dateDeleted` IS NULL)))
WHERE (`elements_sites`.`siteId`=7) AND (`elements`.`id` IS NULL) AND (`elements`.`archived`=FALSE) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)
ORDER BY FIELD(`elements`.`id`,)) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`siteSettingsId`
INNER JOIN `entries` `entries` ON `entries`.`id` = `subquery`.`elementsId`
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `subquery`.`elementsId`) AND (`structureelements`.`structureId` = `subquery`.`structureId`)
ORDER BY FIELD(`elements`.`id`,) in /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Schema.php:676
Stack trace:
#0 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php(1325): yii\db\Schema->convertException(Object(PDOException), 'SELECT `element...')
#1 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php(1186): yii\db\Command->internalExecute('SELECT `element...')
#2 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php(417): yii\db\Command->queryInternal('fetchAll', NULL)
#3 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Query.php(249): yii\db\Command->queryAll()
#4 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/db/Query.php(292): yii\db\Query->all(NULL)
#5 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1853): craft\db\Query->all(NULL)
#6 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Extension/CoreExtension.php(1802): craft\elements\db\ElementQuery->all()
#7 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/helpers/Template.php(148): Twig\Extension\CoreExtension::getAttribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\elements\db\EntryQuery), 'all', Array, 'method', false, false, false, 1)
#8 /Users/dietervanhove/Sites/jims/storage/runtime/compiled_templates/f7/f7b3a30eff1a740b17a202a0c7789211.php(43): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\elements\db\EntryQuery), 'all', Array, 'method', false, false, false, 1)
#9 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Template.php(393): __TwigTemplate_8077349800ff7de111a57c341166c022->doDisplay(Array, Array)
#10 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Template.php(349): Twig\Template->yield(Array, Array)
#11 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Template.php(364): Twig\Template->display(Array)
#12 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/TemplateWrapper.php(35): Twig\Template->render(Array)
#13 /Users/dietervanhove/Sites/jims/vendor/twig/twig/src/Environment.php(306): Twig\TemplateWrapper->render(Array)
#14 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/View.php(500): Twig\Environment->render('formie/_special...', Array)
#15 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/models/Notification.php(166): craft\web\View->renderTemplate('formie/_special...', Array, 'cp')
#16 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/base/Field.php(1339): verbb\formie\models\Notification->renderTemplate(Array, Array)
#17 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/base/Field.php(578): verbb\formie\base\Field->getEmailHtml(Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification), Object(craft\elements\db\EntryQuery))
#18 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/helpers/Variables.php(357): verbb\formie\base\Field->getValueForVariable(Object(craft\elements\db\EntryQuery), Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification))
#19 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/helpers/Variables.php(343): verbb\formie\helpers\Variables::getParsedFieldValue(Object(verbb\formie\fields\Entries), Object(craft\elements\db\EntryQuery), Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification))
#20 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/helpers/Variables.php(241): verbb\formie\helpers\Variables::getParsedFieldValues(Object(verbb\formie\elements\Form), Object(verbb\formie\elements\Submission), Object(verbb\formie\models\Notification))
#21 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/controllers/SubmissionsController.php(1168): verbb\formie\helpers\Variables::getParsedValue('{timestamp}', Object(verbb\formie\elements\Submission), Object(verbb\formie\elements\Form))
#22 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/controllers/SubmissionsController.php(1136): verbb\formie\controllers\SubmissionsController->_setTitle(Object(verbb\formie\elements\Submission), Object(verbb\formie\elements\Form))
#23 /Users/dietervanhove/Sites/jims/vendor/verbb/formie/src/controllers/SubmissionsController.php(438): verbb\formie\controllers\SubmissionsController->_populateSubmission(Object(verbb\formie\elements\Form))
#24 [internal function]: verbb\formie\controllers\SubmissionsController->actionSubmit()
#25 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#26 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#27 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('submit', Array)
#28 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/Application.php(350): yii\base\Module->runAction('formie/submissi...', Array)
#29 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/Application.php(649): craft\web\Application->runAction('formie/submissi...', Array)
#30 /Users/dietervanhove/Sites/jims/vendor/craftcms/cms/src/web/Application.php(312): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#31 /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#32 /Users/dietervanhove/Sites/jims/web/index.php(17): yii\base\Application->run()
#33 /Users/dietervanhove/.composer/vendor/laravel/valet/server.php(110): require('/Users/dieterva...')
#34 {main}
Additional Information:
Array
(
    [0] => 42000
    [1] => 1064
    [2] => You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`' at line 10
)
 {"memory":28493872,"exception":"[object] (yii\\db\\Exception(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`' at line 10
The SQL being executed was: SELECT `elements`.`id`, `elements`.`canonicalId`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateLastMerged`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`id` AS `siteSettingsId`, `elements_sites`.`siteId`, `elements_sites`.`title`, `elements_sites`.`slug`, `elements_sites`.`uri`, `elements_sites`.`content`, `elements_sites`.`enabled` AS `enabledForSite`, `entries`.`sectionId`, `entries`.`fieldId`, `entries`.`primaryOwnerId`, `entries`.`typeId`, `entries`.`postDate`, `entries`.`expiryDate`, `structureelements`.`root`, `structureelements`.`lft`, `structureelements`.`rgt`, `structureelements`.`level`, `structureelements`.`structureId`
FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `siteSettingsId`, `structureelements`.`structureId`
FROM `elements` `elements`
INNER JOIN `entries` `entries` ON `entries`.`id` = `elements`.`id`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `elements`.`id`) AND (EXISTS (SELECT *
FROM `structures` use index(primary)
WHERE (`id` = `structureelements`.`structureId`) AND (`dateDeleted` IS NULL)))
WHERE (`elements_sites`.`siteId`=7) AND (`elements`.`id` IS NULL) AND (`elements`.`archived`=FALSE) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)
ORDER BY FIELD(`elements`.`id`,)) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`siteSettingsId`
INNER JOIN `entries` `entries` ON `entries`.`id` = `subquery`.`elementsId`
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `subquery`.`elementsId`) AND (`structureelements`.`structureId` = `subquery`.`structureId`)
ORDER BY FIELD(`elements`.`id`,) at /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Schema.php:676)
[previous exception] [object] (PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')) `subquery`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`' at line 10 at /Users/dietervanhove/Sites/jims/vendor/yiisoft/yii2/db/Command.php:1320)"} 
2024-11-12 15:52:04 [web.INFO] [application] Request context:
```

**Comment by engram-design** [comment]
This seems related to maybe an Entries field in your form, and sending a notification. The actual error is:

```
ORDER BY FIELD(`elements`.`id`,)) `subquery`
```

While I can't replicate this, I think it should be fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.9"`.

**Comment by engram-design** [comment]
Fixed in [3.0.10](https://github.com/verbb/formie/releases/tag/3.0.10)
